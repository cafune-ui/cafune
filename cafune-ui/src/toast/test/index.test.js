import Toast from '../';
import { render, shallow } from 'enzyme';

function timer(delay = 0) {
  return new Promise(resolve => {
    setTimeout(resolve, delay);
  });
}

describe('Toast', () => {
  it('should show a different toast & close', async () => {
    const tFn = jest.fn();
    Toast.show({
      content: 'test',
      icon: 'like',
      onClose: tFn
    });
    await timer(2600);
    expect(tFn).toHaveBeenCalled();

    Toast.success({
      content: 'test',
      duration: 200
    });
    await timer(1000);
    expect(document.body.querySelector('.caf-toast').innerHTML).toMatchSnapshot();

    Toast.error({
      content: 'test',
      duration: 200
    });
    await timer(1000);
    expect(document.body.querySelector('.caf-toast').innerHTML).toMatchSnapshot();

    Toast.hide();
  });

  it('should show different icon', async () => {
    const sT = Toast.show({
      content: 'test',
      icon: {
        icon: 'home',
        size: '20'
      },
      duration: 200
    });
    await timer(600);

    expect(document.body.querySelector('.caf-toast').innerHTML).toMatchSnapshot();
    Toast.hide();
  });

  it('should loading and change text', async () => {
    const lToast = Toast.loading({ content: 'test' });
    await timer(400);
    expect(
      document.body.querySelector('.caf-toast-notice__content').innerHTML
    ).toEqual('test');
    Toast.setMsg(lToast, 'test text2');
    await timer(200);
    expect(
      document.body.querySelector('.caf-toast-notice__content').innerHTML
    ).toEqual('test text2');
    Toast.hide(lToast);
    await timer(200);
    expect(document.body.querySelector('.caf-toast').innerHTML).toEqual('');
  });
});
