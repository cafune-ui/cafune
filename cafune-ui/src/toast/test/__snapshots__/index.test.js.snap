// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should show a different toast & close 1`] = `"<div class=\\"caf-toast-notice caf-toast-fade caf-toast-fade-enter-active caf-toast-fade-leave-active caf-toast-fade-leave\\"><i class=\\"caf-icons caf-icon caf-icon-success\\" style=\\"font-size: 42px;\\"></i><p class=\\"caf-toast-notice__content\\">test</p></div>"`;

exports[`Toast should show a different toast & close 2`] = `"<div class=\\"caf-toast-notice caf-toast-fade caf-toast-fade-enter-active caf-toast-fade-leave-active caf-toast-fade-leave\\"><i class=\\"caf-icons caf-icon caf-icon-error\\" style=\\"font-size: 42px;\\"></i><p class=\\"caf-toast-notice__content\\">test</p></div>"`;

exports[`Toast should show different icon 1`] = `"<div class=\\"caf-toast-notice caf-toast-fade caf-toast-fade-enter-active caf-toast-fade-leave-active caf-toast-fade-leave\\"><i class=\\"caf-icons caf-icon caf-icon-home\\" style=\\"font-size: 20;\\"></i><p class=\\"caf-toast-notice__content\\">test</p></div>"`;
